src/scalib_ext/scalib/src/sasca/factor_graph.rs

```diff
@@ -201,6 +200,8 @@ pub enum FGError {
     NoEdge { var: String, factor: FactorId },
     #[error("Failure at factor {0}. Expected result {1}, got {2}.")]
     CheckFail(String, PublicValue, PublicValue),
+    #[error("")]
+    InvalidGenericFactorAssignment(String),
```

Comment:
This should be a different case for each distinct error message, avoiding the need to `format!` the errors.